lib/netbox_client_ruby/entities.rb

```diff
@@ -75,6 +75,14 @@ def path(path = nil)
       end
     end
 
+    def find_by(attributes)
+      filter(attributes).find do |netbox_object|
+        attributes.all? do |filter_key, filter_value|
```

Comment:
I suspect we have to add special magic for custom fields. When you filter for custom fields, the `filter_key` is `cf_<something>`.

But the `netbox_object` has no field `cf_<something>`, it only knows `custom_field['<something>']`.

Reply:
Another solution could be that `find_by` has two arguments:

```ruby
    def find_by(filter, attributes == nil)
      attributes == filter if attributes.nil?
      filter(filter).find do |netbox_object|
        attributes.all? do |filter_key, filter_value|
          netbox_object.public_send(filter_key) == filter_value
        end
      end
    end
```

Now,
When `attributes` is nil, then it works the same as you implemented it.
When `attributes` is defined, then it will be used for the final comparison.

It's not as nice, but allows for the flexibility needed when the attributes for filtering (on the remote side) differ from the attributes to double-check locally.

Reply:
Making a bit of magic for the custom fields doesn't seem to much of a hassle to me.

For the alternative implementation, it would help to see a usage example for it.

Reply:
Here's a sample implementation:

```ruby
# Replace
netbox_object.public_send(filter_key) == filter_value

# By
if filter_key.start_with?('cf_')
  netbox_object.custom_field[filter_key.sub('cf_', '')] == filter_value
else
  netbox_object.public_send(filter_key) == filter_value
end
```